### PR TITLE
Replace black/white with allow/deny lists

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
**Description**

Replace white/black list usage with allow/deny lists.

### CODE REVIEW

File: beeline-core/src/test/java/io/honeycomb/beeline/tracing/propagation/ServletFilterPathMatchingTest.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public void destroy() throws Exception {
     }
 
     @Test
-    public void whenNoBlackOrWhitelisting_thenAllSpansCollected() throws Exception {
+    public void whenNoBlackOrAllowlisting_thenAllSpansCollected() throws Exception {
```
Comment (bdarfler) [inline beeline-core/src/test/java/io/honeycomb/beeline/tracing/propagation/ServletFilterPathMatchingTest.java:51]: Ooh, I think one "Black" slipped through here.

Comment (MikeGoldsmith) [reply beeline-core/src/test/java/io/honeycomb/beeline/tracing/propagation/ServletFilterPathMatchingTest.java]: Fixed 👍

**Comment by vreynolds** [comment]
Is this changing URL paths? (do we need a major version bump)

**Comment by MikeGoldsmith** [comment]
> Is this changing URL paths? (do we need a major version bump)

The paths that are changing are for tests only. The only material change to an end user is a method description change and a debug log wording. The remainder of the changes is all internal test stuff (method names, param names, test controller paths, etc) 😄

**Review by vreynolds — APPROVED** [review]

**Pull request merged**